client/src/js/statsTable.js

```diff
@@ -1,108 +1,44 @@
 let currentSortingColumn = 1;
 let isAscending = false;
-let sortedData;
 
-sortedData = sortData();
-createAndAddTable(sortedData);
+var sortedData = sortData();
+updateTable(sortedData);
 
-function sortData(col = 1, isAscending = false) {
-  if (col === 0 || col >= responseData.data[0].length) {
-    console.log('bad column. aborting sortData()!');
-    return;
-  }
-  
-  let runOnce = false;
-  var sortedData = responseData.data.slice(1).sort((a, b) => {
-    
-    if (isAscending) {
-      if (!runOnce) console.log('sortData() sorting by ascending');
-      runOnce = true;
+function updateTable(data) {
+  var html = nunjucks.render('statsTable.html', {
+    table: data,
+    currentSortingColumn: currentSortingColumn,
+    isAscending: isAscending
+  });
+  var table = document.querySelector('#statsTable');
+  table.innerHTML = html;
+}
 
-      return a[col] - b[col];
-    }
-    else {
-      if (!runOnce) console.log('sortData() sorting by descending');
-      runOnce = true;
-      
-      return b[col] - a[col];
+function sortData() {
+  var sortedData = responseData.data.slice(1).sort((a, b) => {
+    a = a[currentSortingColumn];
+    b = b[currentSortingColumn];
+    if (a == b) {
+      return 0;
+    } else if ((isAscending && (a > b)) || ((!isAscending) && (b > a))) {
+      return 1;
+    } else {
+      return -1;
     }
   })
-
   return [responseData.data[0], ...sortedData];
 }
 
-function createAndAddTable(data) {
-  let statsTable;
-  if (statsTable = document.querySelector('#statsTable')) {
-    document.body.removeChild(statsTable);
-  }
-
-  let table = document.createElement("table");
-  table.id = "statsTable";
-  addDataToTable(table, data)
-
-  document.body.appendChild(table);
-
-  console.log('created new table');
-}
-
-function addDataToTable(table, data) {
-  createAndAddThRow(table, data[0]);
-
-  for (let row = 1; row < data.length; row++) {
-    createAndAddTdRow(table, data[row]);
-  }
-}
-
-function createAndAddThRow(table, thContents) {
-  newRow = document.createElement('tr');
-  for (let thContent of thContents) {
-    newRow.appendChild(createCell('th', thContent));
-  }
-
-  newRow.addEventListener('click', handleSort);
-
-  table.appendChild(newRow);
-}
-
-function createAndAddTdRow(table, tdContents) {
-  newRow = document.createElement('tr');
-  for (let tdContent of tdContents) {
-    newRow.appendChild(createCell('td', tdContent));
-  }
-  table.appendChild(newRow);
-}
-
-function createCell(tdOrTh, content) {
-  newCell = document.createElement(tdOrTh);
-  newCell.innerText = content
-  return newCell;
-}
-
-
-function handleSort(evt) {
-  let targetIndex = getIndexFromInnerTextTarget(evt.target.innerText);
-  if (targetIndex === 0 || targetIndex >= responseData.data[0].length) {
-    return;
-  }
-
+function handleSort(th) {
+  var targetIndex = Array.prototype.indexOf.call(th.parentNode.children, th);
```

Comment:
Index via the DOM rather than using the content of the tag and searching in the data. This allows for multiple columns with the same name.